backend/routes/users.js

```diff
@@ -0,0 +1,75 @@
+const router = require('express').Router();
+let User = require('../models/user.model');
+
+/**
+ * Get all users
+ */
+router.route('/').get((req, res) => {
+    User.find()
+        .then(users => res.json(users))
+        .catch(err => res.status(400).json('Error: ' + err));
+});
+
+/**
+ * Get user by ID
+ */
+router.route('/:id').get((req, res) => {
+    User.findById(req.params.id)
+        .then(user => res.json(user))
+        .catch(err => res.status(400).json('Error: ' + err));
+});
+
+/**
+ * Delete a user by ID
+ */
+router.route('/:id').delete((req, res) => {
+    User.findByIdAndDelete(req.params.id)
+        .then(user => res.json(`User deleted`))
+        .catch(err => res.status(400).json('Error: ' + err));
+});
+
+/**
+ * Update a user
+ */
+router.route('/update/:id').post((req, res) => {
+    User.findById(req.params.id)
+        .then(user => {
+            user.username = req.body.username;
+            user.email = req.body.email;
+            user.password = req.body.password;
+            user.firstName = req.body.firstName;
+            user.lastName = req.body.lastName;
+            user.program = req.body.program;
+
+            user.save()
+                .then(() => res.json(`User ${user.email} updated`))
+                .catch(err => res.status(400).json('Error: ' + err));
+        })
+        .catch(err => res.status(400).json('Error: ' + err));
+});
+
+/**
+ * Add a user
+ */
+router.route('/add').post((req, res) => {
+    const username = req.body.username;
```

Comment:
if these variables are only use in the newUser const (line 62), we should put the req.body.username directly in line 63

Reply:
resolving as this is outdated